Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -0,0 +1,158 @@
+  // The attribute index is taken from the node render resources, which may have added some attributes of its own.
+  var attributeIndex = renderResources.attributeIndex;
+  var index;
+  var customAttributeInitializationLines = [];
+  for (var i = 0; i < primitive.attributes.length; i++) {
+    var attribute = primitive.attributes[i];
+    if (attribute.semantic !== VertexAttributeSemantic.POSITION) {
+      index = attributeIndex++;
+    } else {
+      index = 0;
+    }
+    processAttribute(
+      renderResources,
+      attribute,
+      index,
+      customAttributeInitializationLines
+    );
+  }
+
+  var shaderBuilder = renderResources.shaderBuilder;
+
+  // add a function to initialize varyings for custom attributes.
+  // for example "v_custom_attribute = a_custom_attribute;""
+  if (customAttributeInitializationLines.length > 0) {
+    shaderBuilder.addDefine(
+      "HAS_CUSTOM_ATTRIBUTES",
+      undefined,
+      ShaderDestination.VERTEX
+    );
+    var varyingFunctionLines = [].concat(
+      "void initializeCustomAttributes()",
+      "{",
+      customAttributeInitializationLines,
+      "}"
+    );
+    shaderBuilder.addVertexLines(varyingFunctionLines);
+  }
+
+  if (primitive.primitiveType === PrimitiveType.POINTS) {
+    shaderBuilder.addDefine("PRIMITIVE_TYPE_POINTS");
+  }
+
+  shaderBuilder.addVertexLines([GeometryStageVS]);
+  shaderBuilder.addVarying("vec3", "v_positionEC");
+};
+
+function processAttribute(
+  renderResources,
+  attribute,
+  attributeIndex,
+  customAttributeInitializationLines
+) {
+  var semantic = attribute.semantic;
+  var setIndex = attribute.setIndex;
+  var attributeType = attribute.type;
+
+  var shaderBuilder = renderResources.shaderBuilder;
+
+  var variableName;
+  var varyingName;
+  var glslType = AttributeType.getGlslType(attributeType);
+
+  if (defined(semantic)) {
+    variableName = VertexAttributeSemantic.getVariableName(semantic, setIndex);
+    varyingName = "v_" + variableName;
+
+    switch (semantic) {
+      case VertexAttributeSemantic.NORMAL:
+        shaderBuilder.addDefine("HAS_NORMALS");
+        break;
+      case VertexAttributeSemantic.TANGENT:
+        shaderBuilder.addDefine("HAS_TANGENTS");
+        break;
+      case VertexAttributeSemantic.TEXCOORD:
+        shaderBuilder.addDefine("HAS_TEXCOORD_" + setIndex);
+        break;
+      case VertexAttributeSemantic.COLOR:
+        shaderBuilder.addDefine("HAS_VERTEX_COLORS");
+        break;
+      case VertexAttributeSemantic.FEATURE_ID:
+        shaderBuilder.addDefine("HAS_FEATURE_ID");
+        break;
+    }
+  }
+
+  var vertexAttribute = {
+    index: attributeIndex,
+    vertexBuffer: attribute.buffer,
+    componentsPerAttribute: AttributeType.getComponentsPerAttribute(
+      attributeType
+    ),
+    componentDatatype: attribute.componentDatatype,
+    offsetInBytes: attribute.byteOffset,
+    strideInBytes: attribute.byteStride,
+    normalize: attribute.normalized,
+  };
```

Comment:
In some rare cases a vertex attribute does not have a vertex buffer but instead has a constant value. This can happen when a glTF accessor does not have a buffer view and the values are expected to be filled with zeros.

That's what `attribute.constant` is for. I think you can just pass in `value: attribute.constant` and it should work but double check what `VertexArray` is doing.